WordPress/src/main/java/org/wordpress/android/sharedlogin/resolver/SharedLoginResolver.kt

```diff
@@ -96,4 +137,33 @@ class SharedLoginResolver @Inject constructor(
             startActivity(mainActivityIntent)
         }
     }
+
+    private fun copySitesWithIndexes(sites: List<SiteModel>) {
+        val db = WellSql.giveMeWritableDb()
```

Comment:
Should we wrap this DB transaction into another class so that the Resolver doesn't know implementation details of our data layer?

Reply:
Done in ResolverUtility, also added some unit test coverage.